modules/core/test/test_opencl.cpp

```diff
@@ -188,4 +188,133 @@ TEST(OCL_OpenCLExecutionContext, ScopeTest)
     executeUMatCall();
 }
 
+TEST(OCL_OpenCLExecutionContext, reuseExecContext)
+{
+    bool useOCL = cv::ocl::useOpenCL();
+
+    OpenCLExecutionContext initial_ctx = OpenCLExecutionContext::getCurrent();
+
+    if (!useOCL)
+    {
+        ASSERT_TRUE(initial_ctx.empty());  // Other tests should not broke global state
+        throw SkipTestException("OpenCL is not available / disabled");
+    }
+
+    ASSERT_FALSE(initial_ctx.empty());
+
+    try
+    {
+        void *p1, *p2, *p3;
+        {
+            ocl::Context context1 = ocl::Context::create(":GPU:1");
```

Comment:
> `:GPU:1`

Please use `:GPU:0`, so this can be tested on CI with one GPU too.

Reply:
Can do. However...in this same test_opencl.cpp file...I copied verbatim that code. So if you are concerned about this for my two test cases, then your concern should apply to the other two test cases that do the same thing.

Would you like me to fix those other two test cases while I'm in this PR?

Reply:
`:GPU:1` requires extra unnecessary requirements to run this test (to have dual GPU configurations at least).

> other two test cases

I will take a look on them once again a bit later (not in the scope of this PR)